cns/service/main.go

```diff
@@ -1037,6 +1078,91 @@ func main() {
 	logger.Close()
 }
 
+// Poll CRD until it's set and update PluginManager
+func pollNodeInfoCRDAndUpdatePlugin(ctx context.Context, zlog *zap.Logger, pluginManager *deviceplugin.PluginManager) error {
+	kubeConfig, err := ctrl.GetConfig()
+	if err != nil {
+		logger.Errorf("Failed to get kubeconfig for request controller: %v", err)
+		return errors.Wrap(err, "failed to get kubeconfig")
+	}
+	kubeConfig.UserAgent = "azure-cns-" + version
+
+	clientset, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to build clientset")
+	}
+
+	nodeName, err := configuration.NodeName()
+	if err != nil {
+		return errors.Wrap(err, "failed to get NodeName")
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get node %s", nodeName)
+	}
+
+	// check the Node labels for Swift V2
+	if _, ok := node.Labels[configuration.LabelNodeSwiftV2]; !ok {
+		zlog.Info("Node is not labeled for Swift V2, skipping polling nodeinfo crd")
+		return nil
+	}
+
+	directcli, err := client.New(kubeConfig, client.Options{Scheme: multitenancy.Scheme})
+	if err != nil {
+		return errors.Wrap(err, "failed to create ctrl client")
+	}
+
+	nodeInfoCli := multitenancy.NodeInfoClient{
+		Cli: directcli,
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			zlog.Info("Polling context canceled, exiting")
+			return nil
+		default:
+			// Fetch the CRD status
+			nodeInfo, err := nodeInfoCli.Get(ctx, node.Name)
+			if err != nil {
+				zlog.Error("Error fetching nodeinfo CRD", zap.Error(err))
+				return errors.Wrap(err, "failed to get nodeinfo crd")
+			}
+
+			// Check if the status is set
+			if !reflect.DeepEqual(nodeInfo.Status, mtv1alpha1.NodeInfoStatus{}) && len(nodeInfo.Status.DeviceInfos) > 0 {
```

Comment:
`github.com/google/go-cmp/cmp.Equal` is better than `reflect.DeepEqual` for a lot of reasons. It's a drop-in replacement, and it does the right thing more consistently.

Reply:
updated

Reply:
I made this recommendation prior to learning that cmp was intended for tests only. Given that the intention here is to test whether this type is a zero value, I believe it should be sufficient to test whether `nodeInfo.Status.DeviceInfos != nil && len(nodeInfo.Status.DeviceInfos) > 0`. Second opinion might be nice though. @rbtr ?